app/services/renderer_methods.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# macros to fetch information about invoked jobs
+module RendererMethods
+  extend ActiveSupport::Concern
+
+  def find_job_invocation_by_id(job_id, preload: nil)
+    if (job =  JobInvocation.preload(preload).find_by(id: job_id).first)
+      job
+    else
+      raise ::Foreman::Exception.new(N_("Can't find Job Invocation for an id %s"), job_id)
+    end
+  end
+
+  def load_job_invocations(search, includes: nil, preload: nil)
+    load_resource(klass: JobInvocation, search: search, permission: :view_job_invocations, includes: includes, preload: preload)
+  end
+
+  def search_in_hosts_relation(hosts, search_query)
```

Comment:
Why would I want to do this? Why I haven't used the search directly when I was loading the hosts?
And if you think this is useful, shouldn't it be in core?

Reply:
This method trying to solve problem where I've got a relation from another object - like hosts from task. I cannot to filter these result by another method. Of course, I can use the `search_for` method directly. However, this method is from `scoped_search` plugin and I don't think that is good idea to open (allow in safemode) this kind of method. 

So that's the reason why I wrote this method. I have no opinion in that way.
At least, I moved this method to core (Good idea!). PR from core: https://github.com/theforeman/foreman/pull/7879